router/pkg/config/config.go

```diff
@@ -577,6 +577,19 @@ type WebSocketConfiguration struct {
 	ForwardInitialPayload bool `yaml:"forward_initial_payload" envDefault:"true" env:"WEBSOCKETS_FORWARD_INITIAL_PAYLOAD"`
 	// Authentication configuration for the WebSocket Connection
 	Authentication WebSocketAuthenticationConfiguration `yaml:"authentication,omitempty"`
+	// SetClientInfoFromInitialPayload configuration for the WebSocket Connection
+	SetClientInfoFromInitialPayload WebSocketSetClientInfoFromInitialPayloadConfiguration `yaml:"set_client_info_from_initial_payload"`
+}
+
+type WebSocketSetClientInfoFromInitialPayloadConfiguration struct {
+	// Enabled true if the Router should set the client info from the initial payload of a Subscription Request to the Subgraph
+	Enabled bool `yaml:"enabled" envDefault:"true" env:"WEBSOCKETS_SET_CLIENT_INFO_FROM_INITIAL_PAYLOAD_ENABLED"`
+	// ExportToClientRequestHeaders true if the Router should export the client info to the client request headers
+	ExportToClientRequestHeaders bool `yaml:"export_to_client_request_headers" envDefault:"true" env:"WEBSOCKETS_SET_CLIENT_INFO_FROM_INITIAL_PAYLOAD_EXPORT_TO_CLIENT_REQUEST_HEADERS"`
```

Comment:
Forward might be more meaningful here. We are also already in the context. We can be less verbose with the naming.

```
ForwardHeaders
```